src/pluggy/hooks.py

```diff
@@ -247,15 +247,23 @@ def __call__(self, *args, **kwargs):
             raise TypeError("hook calling supports only keyword arguments")
         assert not self.is_historic()
 
-        if self.spec and self.spec.argnames:
-            notincall = set(self.spec.argnames) - set(kwargs.keys())
-            if notincall:
-                warnings.warn(
-                    "Argument(s) {} which are declared in the hookspec "
-                    "can not be found in this hook call".format(tuple(notincall)),
-                    stacklevel=2,
-                )
-        return self._hookexec(self, self.get_hookimpls(), kwargs)
+        # This is written to avoid expensive operations when not needed.
+        if self.spec:
+            for argname in self.spec.argnames:
```

Comment:
Is looping through the args everytime hoping for none missing from the call faster then just always checking what's missing using the `set` difference?

Seems like in average case this will be slower (assuming calls are written correctly most of the time)?
I'm not actually sure I can think of a case where this is faster?
I don't think a `for` loop will ever be faster then a `set` difference but I could be wrong?

Reply:
Here is an unscientific benchmark (Python 3.8, Archlinux), also added a variant which uses `issubset`. Only checks the happy path.

```py
def old(argnames, kwargs):
    if argnames:
        notincall = set(argnames) - kwargs.keys()
        if notincall:
            pass

def old_subset(argnames, kwargs):
    if argnames:
        if not set(argnames).issubset(kwargs.keys()):
            pass

def new(argnames, kwargs):
    for argname in argnames:
        if argname not in kwargs:
            break

import timeit
kwargs = {'a': 0, 'b': 1, 'c': 2, 'd': 3, 'e': 4}
argnames = list(kwargs)
print("old:       ", timeit.timeit("old(argnames, kwargs)", "from __main__ import old, argnames, kwargs"))
print("old_subset:", timeit.timeit("old_subset(argnames, kwargs)", "from __main__ import old_subset, argnames, kwargs"))
print("new:       ", timeit.timeit("new(argnames, kwargs)", "from __main__ import new, argnames, kwargs"))
```

Output:

```
old:        0.7920419139554724
old_subset: 0.6716860989108682
new:        0.2587350399699062
```

Reply:
> (Python 3.8, Archlinux)

You're of my kind :smile_cat: 

>new:        0.2587350399699062

So slick; I guess `for` the win :surfing_man: